Sources/SwiftDocC/Model/Identifier.swift

```diff
@@ -96,6 +96,15 @@ public struct ResolvedTopicReference: Hashable, Codable, Equatable, CustomString
     static func purgePool(for bundleIdentifier: String) {
         sharedPool.sync { $0.removeValue(forKey: bundleIdentifier) }
     }
+    
+    /// Enables reference caching for any identifiers created with the given bundle identifier.
+    static func enableReferenceCaching(for bundleIdentifier: ReferenceBundleIdentifier) {
+        sharedPool.sync { sharedPool in
+            if !sharedPool.keys.contains(bundleIdentifier) {
+                sharedPool[bundleIdentifier] = [:]
```

Comment:
Minor thing here for legibility, which could be left as a future improvement: since the behavior of cache is getting more complicated, it would be nice to refactor into its own type that exposes clearer operations like `enableCaching(forBundleIdentifier:)`. It's not immediately obvious what `sharedPool[bundleIdentifier] = [:]` does, and it looks like we're actually clearing a dictionary that potentially had values before.

Reply:
> Minor thing here for legibility, which could be left as a future improvement: since the behavior of cache is getting more complicated, it would be nice to refactor into its own type that exposes clearer operations like enableCaching(forBundleIdentifier:)

I agree that would be good but I think we should leave that for the more comprehensive change we'll do here in the future (making the cache non-global). My general goal here was to make the change minimal and targeted to reduce risk for landing this on 5.7.

> it looks like we're actually clearing a dictionary that potentially had values before.

The if statement should prevent this, right? That was the intention.

Reply:
Ah right yes, I meant more that the reason we're initializing an empty dictionary here is so that 188 is not a no-op. Generally, I'd say relying on the presence of a key being associated with an empty value is brittle, which is why we could formalise it in a wrapper time. But definitely for a future PR :)

Reply:
Yeah agreed with all of that 👍🏻